streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -0,0 +1,353 @@
+
+        assertThat(testStreamThread.assignedTopicPartitions.get(FIRST_UPDATE), equalTo(expectedFirstAssignment));
+        assertThat(testStreamThread.assignedTopicPartitions.get(SECOND_UPDATE), equalTo(expectedSecondAssignment));
+    }
+
+
+    @Test
+    public void testShouldReadFromRegexAndNamedTopics() throws Exception {
+
+        String topic1TestMessage = "topic-1 test";
+        String topic2TestMessage = "topic-2 test";
+        String topicATestMessage = "topic-A test";
+        String topicCTestMessage = "topic-C test";
+        String topicYTestMessage = "topic-Y test";
+        String topicZTestMessage = "topic-Z test";
+
+
+        final Serde<String> stringSerde = Serdes.String();
+
+        Properties streamsConfiguration = getStreamsConfig();
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        KStream<String, String> pattern1Stream = builder.stream(Pattern.compile("topic-\\d"));
+        KStream<String, String> pattern2Stream = builder.stream(Pattern.compile("topic-[A-D]"));
+        KStream<String, String> namedTopicsStream = builder.stream(TOPIC_Y, TOPIC_Z);
+
+        pattern1Stream.to(stringSerde, stringSerde, DEFAULT_OUTPUT_TOPIC);
+        pattern2Stream.to(stringSerde, stringSerde, DEFAULT_OUTPUT_TOPIC);
+        namedTopicsStream.to(stringSerde, stringSerde, DEFAULT_OUTPUT_TOPIC);
+
+        // Remove any state from previous test runs
+        IntegrationTestUtils.purgeLocalStreamsState(streamsConfiguration);
+
+        KafkaStreams streams = new KafkaStreams(builder, streamsConfiguration);
+        streams.start();
+
+        Properties producerConfig = getProducerConfig();
+
+        produceMessage(TOPIC_1, Arrays.asList(topic1TestMessage), producerConfig);
+        produceMessage(TOPIC_2, Arrays.asList(topic2TestMessage), producerConfig);
+        produceMessage(TOPIC_A, Arrays.asList(topicATestMessage), producerConfig);
+        produceMessage(TOPIC_C, Arrays.asList(topicCTestMessage), producerConfig);
+        produceMessage(TOPIC_Y, Arrays.asList(topicYTestMessage), producerConfig);
+        produceMessage(TOPIC_Z, Arrays.asList(topicZTestMessage), producerConfig);
+
+        Properties consumerConfig = getConsumerConfig();
+
+        List<String> expectedReceivedValues = Arrays.asList(topicATestMessage, topic1TestMessage, topic2TestMessage, topicCTestMessage, topicYTestMessage, topicZTestMessage);
+        List<KeyValue<String, String>> receivedKeyValues = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerConfig, DEFAULT_OUTPUT_TOPIC, 6);
+        List<String> actualValues = new ArrayList<>(6);
+
+        for (KeyValue<String, String> receivedKeyValue : receivedKeyValues) {
+            actualValues.add(receivedKeyValue.value);
+        }
+
+        streams.close();
+        Collections.sort(actualValues);
+        Collections.sort(expectedReceivedValues);
+        assertThat(actualValues, equalTo(expectedReceivedValues));
+    }
+
+    @Test(expected = AssertionError.class)
```

Comment:
I think I understand why now, could you add a TODO marker on it that it should be updated to `expected = TopologyBuilderException` after KAFKA-3708?